### ISSUE TITLE
BUG DUPE/IT DOES NOT WORK

### ISSUE THREAD

**Issue by lollo12345678** [GC-spigot/AdvancedEnchantments#3477]
### Describe the bug

I've been freaking out for hours and hours. I would like the soulbound enchantment that is applied to any object upon death the enchantment is removed and placed back in the player's inventory. This doesn't happen because:
The item is returned to the player's inventory but the enchantment is not removed
In addition, upon death the soulbound item doubles and an identical item is dropped without the soulbound enchantment. Watch the video to better understand I'm going crazy. I also leave here the code of the spell.

soulbound2:
  display: '%group-color%Soulboundtt'
  description: Se crepi hai 100 per cento.
  applies-to: Weapons + Tools + Bows + Armor
  type: DEATH;PASSIVE_DEATH
  group: FABLED
  applies:
    - ALL_TOOLS
    - ALL_SWORD
    - BOW
    - CROSSBOW
    - ALL_ARMOR
  levels:
    '1':
      chance: 100
      cooldown: 0
      effects:
       - REMOVE_ENCHANT:soulbound2
       - KEEP_ON_DEATH

### How to reproduce

Simply put the parameters like in my code as they "Theoretically" should be right with a sword/armor

eh then do /kill or commit suicide

After doing this you will be able to see the same effect that I had as seen in the video, ie the object duplicates and one will be enchanted and the other will not. (I repeat, I would like that upon death the sword goes directly into the player's inventory and the enchantment disappears WITHOUT duplicating it)

### Screenshots / Videos

https://github.com/GC-spigot/AdvancedEnchantments/assets/102913609/23ec07d9-a42f-411c-8035-32d6fb7c42eb

### "/ae plinfo" link

https://paste.md-5.net/iqojaduwib

**Comment by ThomasWega** [comment]
Will be fixed in the next update! :)
